<?xml version="1.0" encoding="utf-8"?>
<robot
  name="agrorob_urdf">
  <link
    name="body">
    <inertial>
      <origin
        xyz="-0.20835 -0.46867 0.1724"
        rpy="0 0 0" />
      <mass
        value="327.77" />
      <inertia
        ixx="16.579"
        ixy="4.8783E-05"
        ixz="0.099256"
        iyy="15.245"
        iyz="-0.036824"
        izz="23.902" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/body.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/body.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="shin_FR">
    <inertial>
      <origin
        xyz="-1.6198E-05 -0.18089 -0.36666"
        rpy="0 0 0" />
      <mass
        value="13.879" />
      <inertia
        ixx="0.17331"
        ixy="4.2107E-05"
        ixz="1.8202E-06"
        iyy="0.16512"
        iyz="-0.012091"
        izz="0.098891" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/shin_FR.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.74902 0.74902 0.74902 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/shin_FR.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="body_shin_FR"
    type="continuous">
    <origin
      xyz="1.5 -0.9 -0.29141"
      rpy="0 0 0.1123" />
    <parent
      link="body" />
    <child
      link="shin_FR" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="wheel_FR">
    <inertial>
      <origin
        xyz="5.0976E-05 -0.006218 -5.2513E-05"
        rpy="0 0 0" />
      <mass
        value="15.288" />
      <inertia
        ixx="0.75083"
        ixy="-3.1731E-05"
        ixz="1.1386E-05"
        iyy="1.2093"
        iyz="2.6415E-05"
        izz="0.75082" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/wheel_FR.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.25098 0.25098 0.25098 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/wheel_FR.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="shin_wheel_FR"
    type="continuous">
    <origin
      xyz="0 0 -0.62885"
      rpy="0 0.34975 0" />
    <parent
      link="shin_FR" />
    <child
      link="wheel_FR" />
    <axis
      xyz="0 1 0" />
  </joint>
  <link
    name="shin_FL">
    <inertial>
      <origin
        xyz="1.6199E-05 0.18089 -0.36666"
        rpy="0 0 0" />
      <mass
        value="13.879" />
      <inertia
        ixx="0.17331"
        ixy="4.2107E-05"
        ixz="-1.8218E-06"
        iyy="0.16512"
        iyz="0.012091"
        izz="0.098891" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/shin_FL.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.74902 0.74902 0.74902 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/shin_FL.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="body_shin_FL"
    type="continuous">
    <origin
      xyz="1.5 0.9 -0.29141"
      rpy="0 0 -0.16579" />
    <parent
      link="body" />
    <child
      link="shin_FL" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="Wheel_FL">
    <inertial>
      <origin
        xyz="5.0976E-05 0.006218 5.2513E-05"
        rpy="0 0 0" />
      <mass
        value="15.288" />
      <inertia
        ixx="0.75083"
        ixy="3.1731E-05"
        ixz="-1.1386E-05"
        iyy="1.2093"
        iyz="2.6414E-05"
        izz="0.75082" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/Wheel_FL.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.25098 0.25098 0.25098 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/Wheel_FL.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="shin_wheel_FL"
    type="continuous">
    <origin
      xyz="0 0 -0.62885"
      rpy="0 -0.94396 0" />
    <parent
      link="shin_FL" />
    <child
      link="Wheel_FL" />
    <axis
      xyz="0 -1 0" />
  </joint>
  <link
    name="shin_RR">
    <inertial>
      <origin
        xyz="-1.6198E-05 -0.18089 -0.36666"
        rpy="0 0 0" />
      <mass
        value="13.879" />
      <inertia
        ixx="0.17331"
        ixy="4.2107E-05"
        ixz="1.8192E-06"
        iyy="0.16512"
        iyz="-0.012091"
        izz="0.09889" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/shin_RR.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.74902 0.74902 0.74902 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/shin_RR.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="body_shin_RR"
    type="continuous">
    <origin
      xyz="-1.5 -0.9 -0.29141"
      rpy="0 0 0.057689" />
    <parent
      link="body" />
    <child
      link="shin_RR" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="wheel_RR">
    <inertial>
      <origin
        xyz="5.0977E-05 -0.006218 -5.2513E-05"
        rpy="0 0 0" />
      <mass
        value="15.288" />
      <inertia
        ixx="0.75083"
        ixy="-3.1732E-05"
        ixz="1.1387E-05"
        iyy="1.2093"
        iyz="2.6414E-05"
        izz="0.75082" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/wheel_RR.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.25098 0.25098 0.25098 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/wheel_RR.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="shin_wheel_RR"
    type="continuous">
    <origin
      xyz="0 0 -0.62885"
      rpy="3.1416 0.71165 3.1416" />
    <parent
      link="shin_RR" />
    <child
      link="wheel_RR" />
    <axis
      xyz="0 1 0" />
  </joint>
  <link
    name="shin_RL">
    <inertial>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <mass
        value="0" />
      <inertia
        ixx="0"
        ixy="0"
        ixz="0"
        iyy="0"
        iyz="0"
        izz="0" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/shin_RL.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/shin_RL.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="body_shin_RL"
    type="continuous">
    <origin
      xyz="-1.5 0.9 -0.29141"
      rpy="0 0 0.10135" />
    <parent
      link="body" />
    <child
      link="shin_RL" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="wheel_RL">
    <inertial>
      <origin
        xyz="5.0976E-05 0.006218 5.2513E-05"
        rpy="0 0 0" />
      <mass
        value="15.288" />
      <inertia
        ixx="0.75083"
        ixy="3.1731E-05"
        ixz="-1.1386E-05"
        iyy="1.2093"
        iyz="2.6414E-05"
        izz="0.75082" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/wheel_RL.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.25098 0.25098 0.25098 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://agrorob_urdf/meshes/wheel_RL.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="shin_wheel_RL"
    type="continuous">
    <origin
      xyz="0 0 -0.62885"
      rpy="-3.1416 0.63844 -3.1416" />
    <parent
      link="shin_RL" />
    <child
      link="wheel_RL" />
    <axis
      xyz="0 -1 0" />
  </joint>
</robot>
